lib/Configuration.php

```diff
@@ -102,4 +104,16 @@ public static function getConfig($category, $key) {
 
 	}
 
+	public static function getVersion() {
+
+		$revisionHashFile = '.git/refs/heads/master';
+
+		if(file_exists($revisionHashFile)) {
+
+			return 'git.' . trim(file_get_contents($revisionHashFile));
```

Comment:
Maybe include `$VERSION` as well?

`return Configuration::$VERSION . ' (' . trim(file_get_contents($revisionHashFile)) . ')';`

=> `2018-06-10 (937ea492716b9251211a2a0f73f3afd1ac871505)`

***

`.git/refs/heads/master` always returns the latest commit in `master`! It wouldn't show branches. Here is an alternative implementation for consideration:

Notice: It uses `exec` to extract the latest commit hash in short form and reads the hash file only if that operation fails. This is purely cosmetic.

```PHP
	public static function getVersion() {
		$headFile = '.git/HEAD';

		$version = Configuration::$VERSION;

		exec('git log --pretty="%h" -n1 HEAD', $output, $return);

		if(!$return) {
			$version .= ' (' . trim($output[0]) . ')';
		} elseif(file_exists($headFile)) {
			$head = trim(explode(':', file_get_contents($headFile))[1]);
			$revisionHashFile = '.git/' . $head;
			if(file_exists($revisionHashFile)) {
				$version .= ' (' . trim(file_get_contents($revisionHashFile)) . ')';
			}
		}

		return $version;
	}
```

Reply:
The reason I decided to read the ```.git/refs/heads/master``` is for simplicity: I know it only shows the ```master``` revision, but if we have a bug report, we only consider it for the version we maitain, that is master. Moreover, a lot of shared hosts do not allow exec, which is why I decided to go for simplicity.

However I believe that we could include the branch name in the version string.

Reply:
Including the branch name sounds like a much simpler alternative.

Using exec was just for cosmetic reasons (the full hash is quite long), as you can see it'll fall back to reading files if exec isn't working. But I'm not fixated on making this more complex than need be, so just leave it out :smile: